crates/bevy_ecs/src/archetype.rs

```diff
@@ -14,6 +14,7 @@ use std::{
 };
 
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+#[repr(transparent)]
```

Comment:
Probably makes sense to add a comment to `ArchetypeId::INVALID` saying that it must be equivalent to setting all bytes of memory to `u8::MAX` (and referencing flush_and_reserve_invalid). Don't want anyone changing this here and then implicitly breaking flush_and_reserve_invalid.